cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -0,0 +1,188 @@
+        navigate_to(appliance.server, 'Server')
+
+        settings_pg = VMwareConsoleSupport(
+            appliance=appliance,
+            console_type='VNC',
+        )
+        settings_pg.update()
+
+
+@pytest.fixture(scope="session")
+def configure_websocket():
+    """
+    Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups.
+    """
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+        yield
+    roles['websocket'] = False
+    logger.info('Disabling the websocket role to avoid intrusive popups')
+    configuration.set_server_roles(**roles)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
+def test_html5_vm_console(appliance, provider, vm_obj, configure_websocket,
+        configure_vmware_console_for_test):
+    """
+    Test the HTML5 console support for a particular provider.
+
+    The supported providers are:
+
+        VMware
+        Openstack
+        RHV
+
+    For a given provider, and a given VM, the console will be opened, and then:
+
+        - The console's status will be checked.
+        - A command that creates a file will be sent through the console.
+        - Using ssh we will check that the command worked (i.e. that the file
+          was created.
+    """
+    console_vm_username = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('username')
+    console_vm_password = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('password')
+
+    vm_obj.open_console(console='VM Console')
+    assert vm_obj.vm_console is not None, 'VMConsole object should be created'
+    vm_console = vm_obj.vm_console
+    try:
+        # If the banner/connection-status element exists we can get
+        # the connection status text and if the console is healthy, it should connect.
+        assert vm_console.wait_for_connect() is True, "VM Console did not reach 'connected' state"
+
+        # Get the login screen image, and make sure it is a jpeg file:
+        screen = vm_console.get_screen()
+        assert imghdr.what('', screen) == 'jpeg'
+
+        assert vm_console.wait_for_text(text_to_find="login:", timeout=200) is not None,\
+            "VM Console didn't prompt for Login"
+
+        # Enter Username:
+        vm_console.send_keys(console_vm_username)
+
+        assert vm_console.wait_for_text(text_to_find="Password", timeout=200) is not None,\
+            "VM Console didn't prompt for Password"
+        # Enter Password:
+        vm_console.send_keys("{}\n".format(console_vm_password))
+
+        time.sleep(5)  # wait for login to complete
```

Comment:
After username and password is entered, sometimes it takes a short while before prompting:

- Either login was successful so I will see command prompt
- Or it failed and it will ask for login.

_try_login() will then be invoked to check if it was successful and if not, attempt for login.  wait_for_text on line 166 will find out if login was successful by trying to read prompt. But only for VMware PhotonOS, for cirros, there is only a "$" sign on prompt line and tesseract can't read it almost always.